src/qililab/platform/platform.py

```diff
@@ -1105,16 +1032,29 @@ def compile(
             ValueError: raises value error if the circuit execution time is longer than ``repetition_duration`` for some qubit.
         """
         # We have a circular import because Platform uses CircuitToPulses and vice versa
-
+        if self.digital_compilation_settings is None:
+            raise ValueError("Cannot compile Qibo Circuit or Pulse Schedule without gates settings.")
         if isinstance(program, Circuit):
-            transpiler = CircuitTranspiler(platform=self)
+            transpiler = CircuitTranspiler(digital_compilation_settings=self.digital_compilation_settings)
             pulse_schedule = transpiler.transpile_circuit(circuits=[program])[0]
         elif isinstance(program, PulseSchedule):
             pulse_schedule = program
         else:
             raise ValueError(
                 f"Program to execute can only be either a single circuit or a pulse schedule. Got program of type {type(program)} instead"
             )
-        return self.compiler.compile(
+        module_and_sequencer_per_bus: dict[str, ModuleSequencer] = {
+            element.bus_alias: ModuleSequencer(module=instrument, sequencer=instrument.get_sequencer(channel))
+            for element in pulse_schedule.elements
+            for instrument, channel in zip(
+                self.buses.get(alias=element.bus_alias).instruments, self.buses.get(alias=element.bus_alias).channels
+            )
+            if isinstance(instrument, QbloxModule)
+        }
+        compiler = PulseQbloxCompiler(
```

Comment:
Here in this compile inside platform, can you confirm that this is Qblox "exclusive" because it is the pulse schedule?

Reply:
Yes, anything that has to do with the `pulse` module, and transpilation from Qibo -> PulseSchedule -> Q1ASM is Qblox specific.